libs/sdk-core/src/persist/transactions.rs

```diff
@@ -345,6 +391,28 @@ fn filter_to_where_clause(
         }
     }
 
+    if let Some(filters) = metadata_filters {
+        if let Ok(map) = serde_json::from_str::<Map<String, Value>>(&filters) {
+            map.iter().for_each(|(key, value)| {
+                let query = match value {
+                    Value::Null => format!("json_extract(metadata, '$.{}') IS NULL", key),
+                    Value::Bool(boolean) => format!(
+                        "json_extract(metadata, '$.{}') = {}", 
+                        key,
+                        *boolean as i32
+                    ),
+                    _ => format!(
+                        "CAST(json_extract(metadata, '$.{}') AS TEXT) = '{}'",
+                        key,
+                        value
+                    )
+                };
+
+                where_clause.push(query);
+            });
+        }
+    }
```

Comment:
Hope you like this approach. After our conversation, I first attempted implementing strings-only, yet realized that if we pass a JSON object for filtering as well (with the desired JSONPath-value fields) I can pretty easily coerce them into the right type! This should support the majority of cases, and is pretty intuitive for the end developer.

Reply:
Sounds good. So something like?
```
{ "path.to.field": true }
```